src/v/bytes/iobuf.cc

```diff
@@ -222,3 +222,61 @@ bool iobuf::operator==(std::string_view o) const {
       });
     return are_equal;
 }
+
+/**
+ * For debugging, string-ize the iobuf in a format like "hexdump -C"
+ *
+ * This is useful if you are handling a parse error and would like to
+ * safely log the unparseable content.  Set an appropriate `limit` to avoid
+ * your log being too verbose.
+ *
+ * @param limit maximum number of bytes to read.
+ * @return a string populated with the following format:
+ *
+00000000 | 7b 22 76 65 72 73 69 6f 6e  22 3a 31 2c 22 6e 61  | {"version":1,"na
```

Comment:
it: in this comment it's two groups of 9 and 7 elements, but it should be 8 and 8 elements